Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift

```diff
@@ -20,14 +20,16 @@ class ProductsFetcherSK1: NSObject {
 
     private var cachedProductsByIdentifier: [String: SK1Product] = [:]
     private let queue = DispatchQueue(label: "ProductsFetcherSK1")
-    private var productsByRequests: [SKRequest: Set<String>] = [:]
+    private var productsByRequests: [SKRequest: ProductRequest] = [:]
     private var completionHandlers: [Set<String>: [(Set<SK1Product>) -> Void]] = [:]
```

Comment:
Outside the scope of this PR, but I I feel like this completion handler should also have an optional `Error` since we only pass an empty set back on error, it would make more sense to me to pass an error back.

Reply:
https://github.com/RevenueCat/purchases-ios/issues/1018

Reply:
I did that in another branch to debug this. But you're right I'll bring it back.

Reply:
Done. Note that in a lot of places I'm simply ignoring the error, which is the same behavior we were doing before (treating it as an empty set).
We can improve this moving forward, but I thought it was out of the scope of this PR.